sdk/cosmos/azure-cosmos/docs/Metrics.md

```diff
@@ -0,0 +1,174 @@
+| cosmos.client.rntbd.endpoints.count             | snapshot | None                   | Current number (snapshot) of service endpoints with any active connections |
+| cosmos.client.rntbd.endpoints.evicted           | snapshot | None                   | Current number (snapshot) of endpoints for which TCP connections were closed |
+| cosmos.client.rntbd.requests.concurrent.count   | snapshot | None                   | Current number (snapshot) of concurrent requests handled per service endpoint |
+| cosmos.client.rntbd.requests.queued.count       | snapshot | None                   | Current number (snapshot) of requests being queued (delayed due to no channel available) per service endpoint |
+| cosmos.client.rntbd.channels.acquired.count     | snapshot | None                   | Current number (snapshot) of new channels (TCP connections) being acquired per service endpoint |
+| cosmos.client.rntbd.channels.available.count    | snapshot | None                   | Current number (snapshot) of existing channels (TCP connections)  per service endpoint |
+
+### System status  metrics (CPU, memory etc.)
+
+| Name                                     | Unit    | Percentiles            | Description                                                  |
+| ---------------------------------------- | ------- | ---------------------- | ------------------------------------------------------------ |
+| cosmos.client.system.avgCpuLoad          | Percent | 95th, 99th + histogram | SDK measures avg. system-wide CPU every 5 seconds. This meter captures the 5-second avg. CPU usage measurements. |
+| cosmos.client.system.freeMemoryAvailable | MB      | None                   | SDK measures free memory available for the process in MB every 5 seconds. This meter captures the 5-second measurements. |
+
+### Request timeline stages
+
+Metrics are available for other stages in the request pipeline as well - but the following stages will be of special interest for monitoring applications:
+
+- channelAcquisitionStarted: Higher values indicate that no channel (TCP connection) is available and a new one needs to be created - either because there is no open connection for the requested replica or because the number of concurrent requests is so high that another connection is needeed
+- transitTime: This is the duration between first byte sent and last byte received - so the time spent in the backend for processing the request + the network turnaround-time. High value with high variance to the backend-reported latency usually indicate either network problems or some resource constraint on the client (or network component like SNAT ports)
+- queued: is the time the requests waits because too many concurrent requests are being processed on a channel (TCP connection). Constantly high values here would indicate that the capacity of the application/service isn't sufficient and would need to be scaled-out.
```

Comment:
`queued ` --> `pipelined`

Reply:
What does queued mean then?

Reply:
Changed queued -> pipelined - but leaving this active to see whether we need/should explain queued a swell.

Reply:
there is no "queued" stage. I think you meant "pipelined" stage

Reply:
No queued stage - but request timeline events are different than stages? https://github.com/Azure/azure-sdk-for-java/blob/ba667dd1e0505fb8e0e0b12bbb16fd5db03a93d9/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestRecord.java#L294-L310

Reply:
ah you are right, there is a queued event in the transport timeline. 

It measure the time between requestRecord created to the time channelAcquisitionStarted. Generally it should be a relatively small value, so the only time have observed high latency is because some thread contention. High latency in this stage does not necessarily caused by too many requests.